docs/appendix/upgrade-notes/4.0-to-4.1.rst

```diff
@@ -5,6 +5,36 @@ From 4.0 to 4.1
    :local:
    :depth: 1
 
+Type hinted functions
+---------------------
+
+The following functions now have their arguments type-hinted, this can cause ``TypeError`` errors. 
+Also some class functions have their return value type hinted and you should update your function definition.
+
+Class function parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* ``Elgg\Forms\StickyForms::clearStickyForm()`` now requires a ``string`` for ``$form_name``
+* ``Elgg\Forms\StickyForms::clearStickyValue()`` now requires a ``string`` for ``$form_name`` and a ``string`` for ``$variable``
+* ``Elgg\Forms\StickyForms::getStickyValue()`` now requires a ``string`` for ``$form_name``, a ``string`` for ``$variable`` and a ``bool`` for ``$filter_result``
+* ``Elgg\Forms\StickyForms::getStickyValues()`` now requires a ``string`` for ``$form_name`` and a ``bool`` for ``$filter_result``
+* ``Elgg\Forms\StickyForms::isStickyForm()`` now requires a ``string`` for ``$form_name``
+* ``Elgg\Forms\StickyForms::makeStickyForm()`` now requires a ``string`` for ``$form_name``
+
+Class function return type
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Lib function parameters
+~~~~~~~~~~~~~~~~~~~~~~~
+
+* ``elgg_clear_sticky_form()`` now requires a ``string`` for ``$form_name``
```

Comment:
this will be breaking... don't do this... cast the variables when calling the class functions